uw-frame-components/portal/widgets/filters.js

```diff
@@ -64,5 +64,20 @@ define(['angular'], function(angular) {
           return -1;
         }
       };
+    })
+    .filter('filterForDateWithYear', function() {
+      return function(date) {
+        var validWithYear = /([12]\d{3}-(0[1-9]|1[0-2])-(0[1-9]|[12]\d|3[01]))/;
```

Comment:
:racehorse: to limit performance implications consider anchoring the start and the end

``` js
/^([12]\d{3}-(0[1-9]|1[0-2])-(0[1-9]|[12]\d|3[01]))$/
```

:notebook: https://momentjs.com/docs/#/parsing/is-valid/ is a wonderful thing